src/services/SearchResults.h

```diff
@@ -126,7 +126,7 @@ class SearchResults
     private:
         union // 8 bytes
         {
-            unsigned char m_vBytes[8];
+            unsigned char m_vBytes[8]{};
```

Comment:
This kept on giving be garbage, this is a lot simpler than doing...
` unsigned char m_vBytes[8] = {static_cast<unsigned char>('\0'), static_cast<unsigned char>('\0'), static_cast<unsigned char>('\0'), static_cast<unsigned char>('\0'), static_cast<unsigned char>('\0'), static_cast<unsigned char>('\0'), static_cast<unsigned char>('\0'), static_cast<unsigned char>('\0'), };`
... in class/structure **initializers** is the standard way of **initializing** something in C++. It makes perfect sense, **initialize** with an **initializer**. The `={}` syntax is not **direct initialization**, that's **copy-initialization**. Why copy when we don't have too? Basically using `={}` means copy the right side into to left instead of just initializing directly.

Whether this is intended to be a block of memory or not, the compiler still thinks it's an array of characters. _unsigned char_ is a _character_ type. We could use std::byte if we aren't to assume it's unsigned char, it has all bit-wise and relational operator overloads already.

Reply:
This is just 8 bytes of RAM. it should be a `uint8_t`.